### PR TITLE
REV: reintroduce csgraph import in scipy.sparse

### PR DESCRIPTION
**Pull request by APwhitehat**
reintroduce csgraph import in scipy.sparse as suggested in #8243

### CODE REVIEW

**Comment by rgommers** [comment]
Looking at the discussion in gh-8280, I'm +1 for merging this.

**Comment by andyfaff** [comment]
Is this not mentioned in the 1.0.0 release notes:

> The deprecated functions sparse.csgraph.cs_graph_components and sparse.linalg.symeig have been removed from scipy.sparse.

How about backport it to the 0.19 and 1.0 maintenance branches, don't include in master, and make sure it's made clear in the release notes for the next release that it's been removed?

**Comment by pv** [comment]
It's about whether scipy.sparse.csgraph is available after "import 
scipy.sparse" or not. I think we don't have good reasons for making it 
unavailable, as long as there are non-deprecated functions in 
scipy.sparse.csgraph.

File: scipy/sparse/__init__.py [hunk 1]
```diff
@@ -237,6 +237,9 @@
 from .extract import *
 from ._matrix_io import *
 
+# For backward compatibility with v0.19. This function is marked as deprecated.
```
Comment (perimosocordiae) [inline scipy/sparse/__init__.py]: `csgraph` is a module, not a function. I'd just leave the bit about back-compat.

**Pull request merged**